src/pypi_attestation_models/_cli.py

```diff
@@ -0,0 +1,302 @@
+        name="sign", help="Sign one or more inputs", parents=[parent_parser]
+    )
+
+    sign_command.add_argument(
+        "--identity-token",
+        type=str,
+        help="Identity token to use",
+        required=False,
+    )
+
+    sign_command.add_argument(
+        "--staging",
+        action="store_true",
+        default=False,
+        help="Use the staging environment",
+    )
+
+    sign_command.add_argument(
+        "files",
+        metavar="FILE",
+        type=Path,
+        nargs="+",
+        help="The file to sign",
+    )
+
+    verify_command = subcommands.add_parser(
+        name="verify",
+        help="Verify one or more inputs",
+        parents=[parent_parser],
+    )
+
+    verify_command.add_argument(
+        "--identity",
+        type=str,
+        required=True,
+        help="Signer identity",
+    )
+
+    verify_command.add_argument(
+        "--staging",
+        action="store_true",
+        default=False,
+        help="Use the staging environment",
+    )
+
+    verify_command.add_argument(
+        "files",
+        metavar="FILE",
+        type=Path,
+        nargs="+",
+        help="The file to sign",
+    )
+
+    inspect_command = subcommands.add_parser(
+        name="inspect",
+        help="Inspect one or more inputs",
+        parents=[parent_parser],
+    )
+
+    inspect_command.add_argument(
+        "--dump-bytes",
+        action="store_true",
+        default=False,
+        help="Dump the bytes of the signature",
+    )
+
+    inspect_command.add_argument(
+        "files",
+        metavar="FILE",
+        type=Path,
+        nargs="+",
+        help="The file to sign",
+    )
+
+    return parser
+
+
+def _die(args: argparse.Namespace, message: str) -> NoReturn:
+    """Handle argument parsing errors and terminate the program.
+
+    Fix up the type hints on our use of `ArgumentParser.error`.
+    """
+    args._parser.error(message)  # noqa: SLF001.
+    raise ValueError("unreachable")
+
+
+def get_identity_token(args: argparse.Namespace) -> IdentityToken:
+    """Generate an Identity Token.
+
+    This method uses the following order of precedence:
+    - A token passed as an argument
+    - An ambient credential
+    - An OAuth-2 flow
+    """
+    # First, check if a token was supplied
+    if args.identity_token:
+        return IdentityToken(args.identity_token)
+
+    # Ambient credential detection
+    oidc_token = detect_credential()
+    if oidc_token is not None:
+        return IdentityToken(oidc_token)
+
+    # Finally, OAuth-2 Flow
+    if args.staging:
+        issuer = Issuer.staging()
+    else:
+        issuer = Issuer.production()
+
+    return issuer.identity_token()
+
+
+def _sign(args: argparse.Namespace) -> None:
+    """Sign the files passed as argument."""
+    try:
+        identity = get_identity_token(args)
+    except IdentityError as identity_error:
+        _die(args, f"Failed to detect identity: {identity_error}")
```

Comment:
I see that `_die` is used to terminate the program here and every place where there are errors or exceptions like `IdentityError`, `ValidationError`, `VerificationError`, etc. But `_die` uses [`ArgumentParser.error()`](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.error), which prints a usage message and is meant for errors related to parsing the command line arguments.

Most of the errors we're handling here and in the rest of the file are not argument parsing errors, rather normal runtime errors (files not existing, validation failing, verification failing, etc). My suggestion would be to change the `_die` uses for normal log + exit statements, in order to avoid printing usage messages on errors that are not due misusing the CLI interface.